galera/src/ist.cpp

```diff
@@ -472,223 +562,236 @@ void galera::ist::Receiver::run()
         bool preload_started(false);
         current_seqno_ = WSREP_SEQNO_UNDEFINED;
 
-        while (true)
         {
-            std::pair<gcs_action, bool> ret;
-
-            if (use_ssl_ == true)
-            {
-                p.recv_ordered(ssl_stream, ret);
-            }
-            else
-            {
-                p.recv_ordered(socket, ret);
-            }
-
-            gcs_action& act(ret.first);
+            SocketWatchdog watchdog([&ssl_stream, &socket, this]() {
+                fprintf(stderr, "SocketWatchdog expired. use_ssl_: %d\n", use_ssl_);
```

Comment:
Shouldn't this print with a normal galera logging macro?

Reply:
fixed